Makefile.am

```diff
@@ -1348,9 +1348,18 @@ HS_MAKEFILE_GHC_SRCS += $(HS_TEST_PROGS:%=%.hs)
 endif
 Makefile.ghc: $(HS_MAKEFILE_GHC_SRCS) Makefile $(HASKELL_PACKAGE_VERSIONS_FILE) \
               | $(built_base_sources) $(HS_BUILT_SRCS)
-	$(GHC) -M -dep-makefile $@ $(DEP_SUFFIXES) $(HFLAGS) $(HFLAGS_DYNAMIC) \
-		-itest/hs \
-		$(HEXTRA_COMBINED) $(HS_MAKEFILE_GHC_SRCS)
+# Run dependency generation for every executable.
+# Up to GHC-9.2 we could perform this in one run of 'ghc -M'.
+# Due to https://gitlab.haskell.org/ghc/ghc/-/issues/25113
+# we have to process every executable manually now.
+# This way, many dependencies in Makefile.ghc are duplicate now
+# but 'make' should cope with this.
+	-mv $@ $@.bak
+	for prog_hs in $(HEXTRA_COMBINED) $(HS_MAKEFILE_GHC_SRCS); do \
+		$(GHC) -M -dep-makefile $@.part $(DEP_SUFFIXES) $(HFLAGS) $(HFLAGS_DYNAMIC) \
+			-itest/hs $$prog_hs ; \
+		cat $@.part >>$@ ; \
```

Comment:
Would it hurt to replace `cat` by `sort -u` as you suggested in #1755?

And is `>>` the automake version of normal shell output redirection `>`, or is this a shell append? May be related to why `mv` is used above, if this a append?